test/e2e/sched/sched_test.go

```diff
@@ -174,5 +181,60 @@ var _ = Describe("[Scheduler] imageReplacement", func() {
 				return true
 			}).WithTimeout(time.Minute * 2).WithPolling(time.Second * 30).Should(BeTrue())
 		})
+		It("should reflect changes in cacheResyncPeriod when configured", func() {
+			nroSchedObj := objects.TestNROScheduler()
+			Expect(e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj)).ToNot(HaveOccurred())
+
+			t := time.Minute * 5
+			nroSchedObj.Spec.CacheResyncPeriod = &metav1.Duration{Duration: t}
+			Expect(e2eclient.Client.Update(context.TODO(), nroSchedObj)).ToNot(HaveOccurred())
+
+			By("checking cacheResyncPeriod under the CR's Status")
+			Expect(e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj)).ToNot(HaveOccurred())
+			Expect(nroSchedObj.Status.CacheResyncPeriod.Duration).To(Equal(nroSchedObj.Spec.CacheResyncPeriod.Duration))
+
+			By("checking cacheResyncPeriod value reflected under the scheduler configMap")
+			cmList := &corev1.ConfigMapList{}
+			Expect(e2eclient.Client.List(context.TODO(), cmList, &client.ListOptions{
+				Namespace: "numaresources",
+			})).ToNot(HaveOccurred())
+
+			var nroschedCM *corev1.ConfigMap
+			for i := 0; i < len(cmList.Items); i++ {
+				if e2eobjects.IsOwnedBy(cmList.Items[i].ObjectMeta, nroSchedObj.ObjectMeta) {
+					nroschedCM = &cmList.Items[i]
+				}
+			}
+			data := nroschedCM.Data[schedstate.SchedulerConfigFileName]
+			Expect(data).ToNot(BeEmpty(), "no data found under %s/%s", nroschedCM.Namespace, nroschedCM.Name)
+
+			schedCfg, err := manifests.DecodeSchedulerConfigFromData([]byte(data))
+			Expect(err).ToNot(HaveOccurred())
+
+			schedProf, pluginConf := findKubeSchedulerProfileByName(schedCfg, schedulerPluginName)
+			Expect(schedProf).ToNot(BeNil())
+			Expect(pluginConf).ToNot(BeNil())
+
+			confObj := pluginConf.Args
+			cfg, ok := confObj.(*pluginconfig.NodeResourceTopologyMatchArgs)
+			Expect(ok).To(BeTrue())
+
+			cacheCfg := time.Duration(cfg.CacheResyncPeriodSeconds) * time.Second
+			Expect(cacheCfg).To(Equal(nroSchedObj.Spec.CacheResyncPeriod.Duration))
```

Comment:
checking the configmap is reflecting the configured value seems good enough.
A future extension (future PR) could also check the `ConfigMap` changed (different resourceversion) and the scheduler pod changed (different UID)